File: ts/512/389.ts
function findTheDifference(s: string, t: string): string {
  const ht = {}
  for (const c of s) {
    if (ht[c]) ht[c] += 1
    else ht[c] = 1
  }
  for (const c of t) {
    if (!ht[c]) return c
    ht[c] -= 1
  }
  return ''
};

File: py3/512/389.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        ht = {}
        ans = ''
        for c in s:
            if c in ht:
                ht[c] += 1
            else:
                ht[c] = 1
        for c in t:
            if c not in ht or ht[c] == 0:
                ans = c
                break
            ht[c] -= 1
        return ans